kvbc/src/v4blockchain/detail/latest_keys.cpp

```diff
@@ -120,6 +123,109 @@ void LatestKeys::handleCategoryUpdates(const std::string& block_version,
+                                     BlockId block_id,
+                                     storage::rocksdb::NativeWriteBatch& write_batch) {
+  ConcordAssertGT(block_id, 0);
+  LOG_DEBUG(V4_BLOCK_LOG, "Reverting keys of block [" << block_id << "] ");
+  auto block_key = v4blockchain::detail::Blockchain::generateKey(block_id);
+  auto prev_block_key = v4blockchain::detail::Blockchain::generateKey(block_id - 1);
+  ConcordAssertEQ(block_key.size(), TIME_STAMP_SIZE);
+  for (const auto& [category_id, updates] : updates.categoryUpdates().kv) {
+    std::visit(
+        [category_id = category_id, &write_batch, &block_key, &prev_block_key, this](const auto& updates) {
+          revertCategoryKeys(block_key, prev_block_key, category_id, updates, write_batch);
+        },
+        updates);
+  }
+}
+
+void LatestKeys::revertCategoryKeys(const std::string& block_version,
+                                    const std::string& prev_block_version,
+                                    const std::string& category_id,
+                                    const categorization::BlockMerkleInput& updates,
+                                    concord::storage::rocksdb::NativeWriteBatch& write_batch) {
+  revertCategoryKeysImp(block_version, prev_block_version, category_id, updates.kv, write_batch);
+  revertDeletedKeysImp(block_version, prev_block_version, category_id, updates.deletes, write_batch);
+}
+
+void LatestKeys::revertCategoryKeys(const std::string& block_version,
+                                    const std::string& prev_block_version,
+                                    const std::string& category_id,
+                                    const categorization::VersionedInput& updates,
+                                    concord::storage::rocksdb::NativeWriteBatch& write_batch) {
+  revertCategoryKeysImp(block_version, prev_block_version, category_id, updates.kv, write_batch);
+  revertDeletedKeysImp(block_version, prev_block_version, category_id, updates.deletes, write_batch);
+}
+
+void LatestKeys::revertCategoryKeys(const std::string& block_version,
+                                    const std::string& prev_block_version,
+                                    const std::string& category_id,
+                                    const categorization::ImmutableInput& updates,
+                                    concord::storage::rocksdb::NativeWriteBatch& write_batch) {
+  revertCategoryKeysImp(block_version, prev_block_version, category_id, updates.kv, write_batch);
+}
+
+template <typename UPDATES>
+void LatestKeys::revertCategoryKeysImp(const std::string& block_version,
+                                       const std::string& prev_block_version,
+                                       const std::string& category_id,
+                                       const UPDATES& updates,
+                                       concord::storage::rocksdb::NativeWriteBatch& write_batch) {
+  static thread_local std::string get_key;
```

Comment:
If we want to use a variable within a thread then why do we want to make it static? Since thread_local will give it storage within the thread. For any variable that we want to use across, we should use static only.
static also provides a linkange-related option. So if we want it a static and not extern linkage, then the thread_local variable by default has a static linkage.

Reply:
the static here is to use a single string instance instead of allocating per call, the thread-local is for safety. 
Since this operation is rare we can omit this optimization if it complicates the code, wdyt?